### DISCUSSION TITLE
how to printf color text ?

### DISCUSSION THREAD

**Discussion by WingDust** [nushell/nushell#10185] [Q&A]
in bash ,zsh 
```sh
 echo -e "\\033[31mcolor text\\033[31m"
# will printf text with color 
```
how to deal with in nushell ?

**Comment by sophiajt** [comment]
it's a similar idea, but using unicode instead of octal escapes:

```
> echo "\u{1b}[31mcolor text\u{1b}[31m"
```

**Comment by fdncred** [comment]
I use `\e` instead of `\\033` or `\u{1b}`
```nushell
print "\e[31mcolor text\e[0m"
```
It's also important here to use double quotes since they interpret escapes, whereas single quotes and backtick quotes do not.

**Comment by ghost** [comment]
another option is to use nushells builtin `ansi` command, which makes it a lot more readable:

```nu
print $"(ansi red)Red text(ansi g) Green text (ansi blue_bold)Blue bold text(ansi reset) \(text in brackets, which is not interpreted as code thanks to the backslash)"

# list all available colors (and shortcuts):
ansi -l
```

**Reply by amtoine** [reply]
i would use the `ansi` command, looks much easier to me :yum:
